include/exec/sequence_senders.hpp

```diff
@@ -0,0 +1,777 @@
+        friend void tag_invoke(set_error_t, _Self&& __self, _Error e) noexcept {
+          stdexec::set_error((_Receiver&&) __self.__op_->__rcvr_, (_Error&&) e);
+        }
+
+        friend void tag_invoke(set_stopped_t, __receiver&& __self) noexcept {
+          auto token = stdexec::get_stop_token(stdexec::get_env(__self.__op_->__rcvr_));
+          if (token.stop_requested()) {
+            stdexec::set_stopped((_Receiver&&) __self.__op_->__rcvr_);
+          } else {
+            stdexec::set_value((_Receiver&&) __self.__op_->__rcvr_);
+          }
+        }
+
+        friend env_of_t<_Receiver> tag_invoke(get_env_t, const __receiver& __self) noexcept(
+          __nothrow_callable<get_env_t, const _Receiver&>) {
+          return stdexec::get_env(__self.__op_->__rcvr_);
+        }
+      };
+
+      struct __t {
+        [[no_unique_address]] _Receiver __rcvr_;
+        [[no_unique_address]] _SourceSender __source_;
+        exec::trampoline_scheduler __trampoline_;
+        std::optional<connect_result_t<__next_on_scheduler_sender, __receiver>> __next_op_;
+
+        void repeat() noexcept {
+          auto __token = stdexec::get_stop_token(stdexec::get_env(__rcvr_));
+          if (__token.stop_requested()) {
+            stdexec::set_stopped((_Receiver&&) __rcvr_);
+            return;
+          }
+          try {
+            auto& __next = __next_op_.emplace(__conv{[&] {
+              return connect(
+                stdexec::on(__trampoline_, set_next(__rcvr_, _SourceSender{__source_})),
+                __receiver{this});
+            }});
+            stdexec::start(__next);
+          } catch (...) {
+            stdexec::set_error((_Receiver&&) __rcvr_, std::current_exception());
+          }
+        }
+
+        friend void tag_invoke(start_t, __t& __self) noexcept {
+          __self.repeat();
+        }
+
+        template <__decays_to<_SourceSender> _Sndr, __decays_to<_Receiver> _Rcvr>
+        explicit __t(_Sndr&& __source, _Rcvr&& __rcvr)
+          : __rcvr_{(_Rcvr&&) __rcvr}
+          , __source_{(_Sndr&&) __source} {
+        }
+      };
+    };
+
+    template <class _SourceSender, class _Receiver>
+    using __operation_t = __t<__operation<__id<_SourceSender>, __id<__decay_t<_Receiver>>>>;
+
+    template <class _Source, class _Env>
+    using __compl_sigs = make_completion_signatures<
+      _Source,
+      _Env,
+      completion_signatures<set_error_t(std::exception_ptr), set_stopped_t()>>;
+
+    template <class _SourceId>
+    struct __sender {
+      using _Source = stdexec::__t<__decay_t<_SourceId>>;
+
+      template <class _Rcvr>
+      using __next_sender = __call_result_t<set_next_t, __decay_t<_Rcvr>&, _Source>;
+
+      template <class _Rcvr>
+      using __next_on_scheduler_sender =
+        __call_result_t<on_t, exec::trampoline_scheduler, __next_sender<_Rcvr>&&>;
+
+      template <class _Rcvr>
+      using __recveiver = typename __operation<__id<_Source>, __id<__decay_t<_Rcvr>>>::__receiver;
```

Comment:
I've found this to be a problematic pattern. You can't compute the receiver type without instantiating the operation state type. But you're using the receiver type below in the constraints of `connect`. This means that, rather than SFINAE away, an invalid call to `connect` will cause a hard error somewhere in the guts of the operation state.

It's usually best to define the receiver type independent of the operation state. And in most cases it doesn't need to depend on the sender type at all.

Reply:
Good catch. I will test and refactor that

Reply:
In case of repeat I found it difficult to not depend on the sender type. I will think about it again.